Source/WebCore/bindings/js/JSDOMConvertStrings.cpp

```diff
@@ -118,4 +121,94 @@ AtomString valueToUSVAtomString(JSGlobalObject& lexicalGlobalObject, JSValue val
     return replaceUnpairedSurrogatesWithReplacementCharacter(WTFMove(string));
 }
 
+String valueToTrustedHTMLString(JSGlobalObject& lexicalGlobalObject, JSValue value, const String& sink, bool nullToEmpty)
+{
+    UNUSED_PARAM(sink);
+    VM& vm = lexicalGlobalObject.vm();
+
+    if (auto* trustedHTML = JSTrustedHTML::toWrapped(vm, value))
+        return trustedHTML->toString();
```

Comment:
We need to find a way to cut down on the boilerplate and repetitiveness of these functions. All these different ones and this line of code is the only difference. There has to be some technique where we don’t just paste the code over and over again. The simplest way might be to have a helper function that takes a lambda for converting the HTML object to a string, or there could be some other technique.

Reply:
I've refactored this to more closely follow the spec shape and as a result this is much more streamlined.